pkgs/development/coq-modules/mathcomp-abel/default.nix

```diff
@@ -0,0 +1,22 @@
+{ coq, mkCoqDerivation, mathcomp, mathcomp-real-closed, lib, version ? null }:
+
+with lib; mkCoqDerivation {
+
+  namePrefix = [ "coq" "mathcomp" ];
```

Comment:
```suggestion
mkCoqDerivation {
  namePrefix = [ "coq" "mathcomp" ];
```

Reply:
Why? I need it for `versions` and `switch below...

Reply:
It is not a good practice to use with over the entire derivation. Apply it where it is necessary.

Reply:
My I ask why this is not a good practice and where it is documented?

Reply:
Because it can overshadow packages especially when you work with python packages so it is just a good practice to only apply it where it is needed. No idea if there is a document about this. Feel free to search yourself.

Reply:
> Because it can overshadow packages especially when you work with python packages so it is just a good practice to only apply it where it is needed.

OK, I see, thanks for the explanation. I did as you said anyway, I may merge if you find nothing more to fix.


> No idea if there is a document about this. Feel free to search yourself.

I did and found nothing so far... And I think it should be. Given you might explain it better than me, could you PR it to nix documentation of `with` expressions? https://nixos.org/manual/nix/stable/#idm140737321917328

Reply:
I am not feeling like bikeshedding over something in the docs right now.